Security Consideration: add Export Vulnerabilities section (#4570)
DevExpress/devextreme-documentation: concepts/Common/Security Considerations/30 Export Vulnerabilities/05 CSV Injection.md

> If you use DataGrid or PivotGrid Export functionality, you should consider a threat of a CSV Injection Attack (also called formula injection attack). It involves the injection of harmful characters that act as a formula and are executed within a computer network. Cell values that begin with the **=, +, -** and **@** characters can initiate an injection attack.

Reviewer: Is CSV the only format available for export in your `DataGrid` and `PivotGrid` products?

> If you use DataGrid or PivotGrid Export functionality

A definite article is missing here. You speak of **the** functionality available where? In `DataGrid` and `PivotGrid`.

> It involves **the** injection of harmful characters that act as a formula and are executed within a computer network.

Do we know of a specific form of such an injection in the way it always takes place? No, we do not. In the current context, it's just **an** injection.

>  Cell values that begin with the **=, +, -** and **@** characters can initiate an injection attack.

An Oxford comma is missing at the end of this enumeration.

Here's my take on this paragraph the way I see it right now. Feel free to correct me if I misinterpreted something:

```suggestion
If you export data from DataGrid or PivotGrid in CSV format, you should take a thread of a CSV Injection Attack (also known as a formula injection attack). Such attacks involve an injection of a malicious character sequence that is interpreted as a formula and executed within a computer network. Cell values that start with the **=**, **+**, **-**, and **@** characters can initiate an injection attack.
```

P.S. I also took the liberty of rephrasing the second sentence a bit for better text flow and clarity.

> If you export data from DataGrid or PivotGrid in CSV format, you should take a thread of a CSV Injection Attack (also known as a formula injection attack). Such attacks involve an injection of a malicious character sequence that is interpreted as a formula and executed within a computer network. Cell values that start with the **=**, **+**, **-**, and **@** characters can initiate an injection attack.

Reviewer 2: "If you export data from DataGrid or PivotGrid in CSV format, you should take a thread of a CSV Injection Attack (also known as a formula injection attack)."

What do you mean by "a thread" in the context of a CSV Injection Attack?

Reviewer: Hello Albert,

My bad, looks like I haven't finished editing this line. Actually, I meant the following instead:

```suggestion
If you export data from DataGrid or PivotGrid in CSV format, you should take a threat of a CSV Injection Attack (also known as a formula injection attack) into account. Such attacks involve an injection of a malicious character sequence that is interpreted as a formula and executed within a computer network. Cell values that start with the **=**, **+**, **-**, and **@** characters can initiate an injection attack.
```

Reviewer 2: Thanks, Dmitry. I edited your new version slightly below:

```suggestion
If you export data from DataGrid or PivotGrid in CSV format, you should take the threat of a CSV Injection Attack (also known as a formula injection attack) into account. Such attacks involve an injection of a malicious character sequence that is interpreted as a formula and executed within a computer network. Cell values that start with the **=**, **+**, **-**, and **@** characters can initiate an injection attack.
```

Reviewer: Thank you very much! I should have done so in the first place...

> When executed, malicious code in a formula can tamper with user data or provide unauthorized access to data and internal resources.

Reviewer 2: ```suggestion
When executed, malicious code in a formula can tamper with user data, or allow unauthorized access to data and internal resources.
```

> When you enter the malicious code into CSV files (in the form of a formula), the formula can be executed. Once executed, the malicious formula can manipulate/delete user data or provide unauthorized access to data/internal resources.

Reviewer: > When you enter **the** malicious code into CSV files (in the form of a formula)

Normally, "code" is uncountable and requires no article unless we have a very good reason for placing an extra emphasis on this word.

> When you enter the malicious code into CSV files (in the form of a formula)

It's strange that we imply that our reader places malicious code into CSV files, moreover, intentionally... Moreover, this sentence repeats the very thing we said in the first paragraph. We can safely delete this sentence and lose nothing.

> manipulate

Normally, we do not use this word in documentation due to its negative connotation. In this case, negative connotation is justified, however, I believe there is a better alternative. The word "tamper", for example, is tightly associated with malicious code and its activity.

```suggestion
When executed, malicious code in a formula can tamper with user data or provide unauthorized access to data and internal resources.
```

P.S. You may consider merging this sentence with the previous paragraph.

> You can encode CSV files to prevent execution of potentially harmful code in them. Pass the [encodeExecutableContent](/Documentation/ApiReference/Common/Object_Structures/ExcelExportDataGridProps/#encodeExecutableContent) option as an argument of the configuration object of the [exportDataGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) or [exportPivotGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportPivotGridoptions) function.

Reviewer 2: ```suggestion
You can encode CSV files to prevent execution of potentially harmful code within them. Pass the [encodeExecutableContent](/Documentation/ApiReference/Common/Object_Structures/ExcelExportDataGridProps/#encodeExecutableContent) option as an argument of the configuration object of the [exportDataGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) or [exportPivotGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportPivotGridoptions) function.
```

> You can encode the CSV files to prevent execution of harmful code in the exported files. Pass the [encodeExecutableContent](/Documentation/ApiReference/Common/Object_Structures/ExcelExportDataGridProps/#encodeExecutableContent) option as an argument of the configuration object of the [exportDataGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) or [exportPivotGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportPivotGridoptions) function.

Reviewer: > You can encode **the** CSV files to prevent execution of harmful code in the exported files.

"CSV files" are plural here and require no article unless we have a very good reason to emphasize them.

> You can encode **the CSV files** to prevent execution of harmful code in **the exported files**.

We speak of the same thing, then why should we repeat ourselves?

I can also note that no code can be executed once we encode all executable fragments in an exported CSV file.

```suggestion
You can encode CSV files to prevent execution of potentially harmful code in them. Pass the [encodeExecutableContent](/Documentation/ApiReference/Common/Object_Structures/ExcelExportDataGridProps/#encodeExecutableContent) option as an argument of the configuration object of the [exportDataGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportDataGridoptions) or [exportPivotGrid](/Documentation/ApiReference/Common/Utils/excelExporter/#exportPivotGridoptions) function.
```
